frigate/http.py

```diff
@@ -517,19 +552,92 @@ def recordings(camera_name):
     )
 
 
-@bp.route("/vod/<year_month>/<day>/<hour>/<camera>")
-def vod(year_month, day, hour, camera):
-    start_date = datetime.strptime(f"{year_month}-{day} {hour}", "%Y-%m-%d %H")
-    end_date = start_date + timedelta(hours=1) - timedelta(milliseconds=1)
-    start_ts = start_date.timestamp()
-    end_ts = end_date.timestamp()
+@bp.route("/<camera>/start/<int:start_ts>/end/<int:end_ts>/clip.mp4")
+@bp.route("/<camera>/start/<float:start_ts>/end/<float:end_ts>/clip.mp4")
+def recording_clip(camera, start_ts, end_ts):
+    download = request.args.get("download", type=bool)
 
-    # Select all recordings where either the start or end dates fall in the requested hour
     recordings = (
         Recordings.select()
         .where(
             (Recordings.start_time.between(start_ts, end_ts))
             | (Recordings.end_time.between(start_ts, end_ts))
+            | ((start_ts > Recordings.start_time) & (end_ts < Recordings.end_time))
+        )
+        .where(Recordings.camera == camera)
+        .order_by(Recordings.start_time.asc())
+    )
+
+    playlist_lines = []
+    clip: Recordings
+    for clip in recordings:
+        playlist_lines.append(f"file '{clip.path}'")
+        # if this is the starting clip, add an inpoint
+        if clip.start_time < start_ts:
+            playlist_lines.append(f"inpoint {int(start_ts - clip.start_time)}")
+        # if this is the ending clip, add an outpoint
+        if clip.end_time > end_ts:
+            playlist_lines.append(f"outpoint {int(end_ts - clip.start_time)}")
+
+    path = f"/tmp/cache/tmp_clip_{camera}_{start_ts}-{end_ts}.mp4"
```

Comment:
I think you should be able to have ffmpeg write the bytes to `pipe:` and read it directly into memory instead of writing to disk as an intermediary. You may even be able to stream the output bytes from ffmpeg directly to the response.

Reply:
I tried that at first, but stdout isn’t seekable, and the MP4 muxer required a seekable output. This writes it to the tmpfs memory file system anyway, so it’s still not really writing it to disk.